src/main/java/duke/Event.java

```diff
@@ -0,0 +1,74 @@
+package duke;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDateTime;
+
+/**
+ * The Event class encapsulates an Event task.
+ */
+public class Event extends Task {
+    /** A specific start and end to the task */
+    protected String at;
+    /** The event duration to be stored in the Event object */
+    protected LocalDateTime event;
+
+    /**
+     * Instantiates an Event object.
+     *
+     * @param description Description of the task.
+     * @param at A specific start and end to the task.
+     * @throws DukeException If at given is in an invalid format.
+     */
+    public Event(String description, String at) throws DukeException {
+        super(description);
+        this.at = at;
+        try {
+            this.event = LocalDateTime.parse(at.substring(1));
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Please input the deadline in yyyy-mm-ddTHours:Minutes:Seconds format. "
```

Comment:
I like how the specific input for the time format is specified for a more user-friendly experience! (but I think 'deadline' should be 'event time' or something similar instead)